mungegithub/github/github.go

```diff
@@ -1004,64 +1022,67 @@ func computeStatus(combinedStatus *github.CombinedStatus, requiredContexts []str
 	}
 }
 
-func (obj *MungeObject) getCombinedStatus() (status *github.CombinedStatus) {
+func (obj *MungeObject) getCombinedStatus() (status *github.CombinedStatus, ok bool) {
 	config := obj.config
-	pr, err := obj.GetPR()
-	if err != nil {
-		return nil
+	pr, ok := obj.GetPR()
+	if !ok {
+		return nil, true
 	}
 	if pr.Head == nil {
 		glog.Errorf("pr.Head is nil in getCombinedStatus for PR# %d", *obj.Issue.Number)
-		return nil
+		return nil, false
 	}
 	// TODO If we have more than 100 statuses we need to deal with paging.
 	combinedStatus, response, err := config.client.Repositories.GetCombinedStatus(config.Org, config.Project, *pr.Head.SHA, &github.ListOptions{})
 	config.analytics.GetCombinedStatus.Call(config, response)
 	if err != nil {
 		glog.Errorf("Failed to get combined status: %v", err)
-		return nil
+		return nil, false
 	}
-	return combinedStatus
+	return combinedStatus, true
 }
 
 // SetStatus allowes you to set the Github Status
-func (obj *MungeObject) SetStatus(state, url, description, context string) error {
+func (obj *MungeObject) SetStatus(state, url, description, context string) bool {
 	config := obj.config
 	status := &github.RepoStatus{
 		State:       &state,
 		TargetURL:   &url,
 		Description: &description,
 		Context:     &context,
 	}
-	pr, err := obj.GetPR()
-	if err != nil {
-		return err
+	pr, ok := obj.GetPR()
+	if !ok {
+		glog.Errorf("Error in SetStatus")
+		return false
 	}
 	ref := *pr.Head.SHA
 	glog.Infof("PR %d setting %q Github status to %q", *obj.Issue.Number, context, description)
 	config.analytics.SetStatus.Call(config, nil)
 	if config.DryRun {
-		return nil
+		return true
 	}
-	_, _, err = config.client.Repositories.CreateStatus(config.Org, config.Project, ref, status)
+	_, _, err := config.client.Repositories.CreateStatus(config.Org, config.Project, ref, status)
 	if err != nil {
 		glog.Errorf("Unable to set status. PR %d Ref: %q: %v", *obj.Issue.Number, ref, err)
 	}
-	return err
+	return false
 }
 
 // GetStatus returns the actual requested status, or nil if not found
-func (obj *MungeObject) GetStatus(context string) *github.RepoStatus {
-	combinedStatus := obj.getCombinedStatus()
-	if combinedStatus == nil {
-		return nil
+func (obj *MungeObject) GetStatus(context string) (*github.RepoStatus, bool) {
+	combinedStatus, ok := obj.getCombinedStatus()
+	if combinedStatus == nil || ok == false {
```

Comment:
Break this up. `if !ok { return false } else if combinedStatus == nil {return true}`   if we already got a `glog` from getCominbedStatus you don't need to do another here.